consensus/module.go

```diff
@@ -75,48 +73,53 @@ type ConsensusModule struct {
 	messagePool map[typesCons.HotstuffStep][]*typesCons.HotstuffMessage
 }
 
-func Create(configPath, genesisPath string, useRandomPK bool) (modules.ConsensusModule, error) {
-	cm := new(ConsensusModule)
-	c, err := cm.InitConfig(configPath)
-	if err != nil {
-		return nil, err
+func Create(runtimeMgr modules.RuntimeMgr) (modules.Module, error) {
+	return new(consensusModule).Create(runtimeMgr)
+}
+
+func (*consensusModule) Create(runtimeMgr modules.RuntimeMgr) (modules.Module, error) {
+	var m *consensusModule
+
+	cfg := runtimeMgr.GetConfig()
+	if err := m.ValidateConfig(cfg); err != nil {
+		return nil, fmt.Errorf("config validation failed: %w", err)
 	}
-	g, err := cm.InitGenesis(genesisPath)
-	if err != nil {
-		return nil, err
+	consensusCfg := cfg.GetConsensusConfig()
+
+	genesis := runtimeMgr.GetGenesis()
+	if err := m.ValidateGenesis(genesis); err != nil {
+		return nil, fmt.Errorf("genesis validation failed: %w", err)
 	}
-	cfg := c.(*typesCons.ConsensusConfig)
-	genesis := g.(*typesCons.ConsensusGenesisState)
-	leaderElectionMod, err := leader_election.Create(cfg, genesis)
+	consensusGenesis := genesis.GetConsensusGenesisState()
+
+	leaderElectionMod, err := leader_election.Create(runtimeMgr)
 	if err != nil {
 		return nil, err
 	}
 
 	// TODO(olshansky): Can we make this a submodule?
-	paceMaker, err := CreatePacemaker(cfg)
+	paceMakerMod, err := CreatePacemaker(runtimeMgr)
 	if err != nil {
 		return nil, err
 	}
 
-	valMap := typesCons.ValidatorListToMap(genesis.Validators)
-	var privateKey cryptoPocket.PrivateKey
-	if useRandomPK {
-		privateKey, err = cryptoPocket.GeneratePrivateKey()
-	} else {
-		privateKey, err = cryptoPocket.NewPrivateKey(cfg.PrivateKey)
-	}
+	valMap := typesCons.ActorListToValidatorMap(consensusGenesis.GetVals())
+
+	privateKey, err := cryptoPocket.NewPrivateKey(consensusCfg.GetPrivateKey())
 	if err != nil {
 		return nil, err
 	}
 	address := privateKey.Address().String()
 	valIdMap, idValMap := typesCons.GetValAddrToIdMap(valMap)
 
-	m := &ConsensusModule{
+	paceMaker := paceMakerMod.(Pacemaker)
+
+	m = &consensusModule{
 		bus: nil,
 
 		privateKey:  privateKey.(cryptoPocket.Ed25519PrivateKey),
-		consCfg:     cfg,
-		consGenesis: genesis,
+		consCfg:     cfg.GetConsensusConfig().(*typesCons.ConsensusConfig),
```

Comment:
Blocker:

We should not cast here because this prevents us from using mocks etc. 

We should use interfaces only

Reply:
We are now using interfaces and I think I found a solution for the problem: "What if there are Consensus modules that have a Pacemaker and others that don't?" (and similar use-cases).